src/migration/1712228361792-PostSlug.ts

```diff
@@ -0,0 +1,15 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class PostSlug1712228361792 implements MigrationInterface {
+  name = 'PostSlug1712228361792';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "post" ADD "slug" text NOT NULL GENERATED ALWAYS AS (trim(BOTH '-' FROM regexp_replace(lower(trim(COALESCE(post.title,'empty-title')||'-'||post.id)), '[^a-z0-9\\\\-_]+', '-', 'gi'))) STORED;`,
```

Comment:
if title is empty let's fallback to just the id. but i think we should check if title is empty after stripping all the unique characters

Reply:
The issue is that we can't use concat_ws which does exactly that because it's not immutable.
So we have to do individual fallbacks, can maybe fallback to empty space which will get cleaned?

Reply:
This field also needs tests to make sure it handles all the special cases. it's high impact change so we must test it

Reply:
> The issue is that we can't use concat_ws which does exactly that because it's not immutable.
So we have to do individual fallbacks, can maybe fallback to empty space which will get cleaned?
Not sure i understand but I'll see in the test case

Reply:
@idoshamun Made it so it render empty title rather.
Checked it works and still adds post id as fallback.

Will add some tests as well :)

Reply:
@idoshamun Added some tests with the examples from google.
Let me know if that works?

Note we don't do the full UTF-8 as we only rely on PG functions for now.
(I personally think this nuance is good as it allows faster shipping + less overhead/triggers) and our posts generally don't go that wild.